vizro-core/src/vizro/figures/kpi_cards.py

```diff
@@ -1,144 +1,8 @@
-            - reference: `reference_column` aggregated by `agg_func`.
-            - delta: Difference between `value` and `reference`.
-            - delta_relative: Relative difference between `value` and `reference`.
-
-            **Common examples include:**
-
-             - "{value}": Displays the raw value.
-             - "${value:0.2f}": Formats the value as a currency with two decimal places.
-             - "{value:.0%}": Formats the value as a percentage without decimal places.
-             - "{value:,}": Formats the value with comma as a thousands separator.
-
-        reference_format: Format string to be applied to the reference. For more details on possible placeholders, see
-            docstring on `value_format`. Defaults to "{delta_relative:+.1%} vs. reference ({reference})".
-        agg_func: String function name to be used for aggregating the data. Common options include
-            "sum", "mean" or "median". Default is "sum". For more information on possible functions, see
-            https://stackoverflow.com/questions/65877567/passing-function-names-as-strings-to-pandas-groupby-aggregrate.
-        title: KPI title displayed on top of the card. If not provided, it defaults to the capitalized `value_column`.
-        icon: Name of the icon from the [Google Material Icon Library](https://fonts.google.com/icons) to be displayed
-            on the left side of the KPI title. If not provided, no icon is displayed.
-
-    Raises:
-        UserWarning: If `value_format` or `reference_format` is provided, a warning is raised to make aware
-            that only trusted user input should be provided.
-
-    Returns:
-        A Dash Bootstrap Components card (`dbc.Card`) containing the formatted KPI value and reference.
-
-    """
-    title = title or f"{agg_func} {value_column}".title()
-    value, reference = data_frame[[value_column, reference_column]].agg(agg_func)
-    delta = value - reference
-    delta_relative = delta / reference if reference else np.nan
-
-    header = dbc.CardHeader([html.P(icon, className="material-symbols-outlined") if icon else None, html.H2(title)])
-    body = dbc.CardBody(
-        value_format.format(value=value, reference=reference, delta=delta, delta_relative=delta_relative)
-    )
-    footer = dbc.CardFooter(
-        [
-            html.Span(
-                "arrow_circle_up" if delta > 0 else "arrow_circle_down" if delta < 0 else "arrow_circle_right",
-                className="material-symbols-outlined",
-            ),
-            html.Span(
-                reference_format.format(value=value, reference=reference, delta=delta, delta_relative=delta_relative)
-            ),
-        ],
-        className="color-pos" if delta > 0 else "color-neg" if delta < 0 else "",
-    )
-    return dbc.Card([header, body, footer], className="card-kpi")
+kpi_card = capture("figure")(kpi_card)
+kpi_card_reference = capture("figure")(kpi_card_reference)
```

Comment:
@antonymilne - do you think it looks really weird if I actually do it the other way around? Currently we have the pure function code live in figure.undecorated.kpi_cards and in figure.kpi_cards we just wrap the function as above.

However, we could also have it the other way around. So let the pure function code live in figure.kpi_cards and in figure.undecorated.kpi_cards, we do this instead:

```
kpi_card = kpi_card.__wrapped__
kpi_card_reference = kpi_card.__wrapped__
```

I know it's less intuitive but the current implementation messes up our API reference as it can't read it 😬

Reply:
Fine by me 👍 I'm surprised the API docs can't handle this though, would be curious to understand why 🤔 Maybe it's because of a name clash and it would work if you imported the unwrapped ones explicitly as `unwrapped_kpi_card`?

Reply:
Hmm.. not sure why to be honest. I've tried to rename it as you've suggested but it didn't work. 

Read through this:
- https://github.com/mkdocstrings/mkdocstrings/issues/162 
- https://mkdocstrings.github.io/troubleshooting/#my-wrapped-function-shows-documentationcode-for-its-wrapper-instead-of-its-own

However, I saw that we do use `functools.wrap` already and still does not work 🤔 I didn't spend much more time on this as the alternative solution above works, but if it's really important to understand and get the other way around working I can take a look again after I've taken a look at the CSS stuff.

Reply:
Yes indeed, I was expecting `functools.wrap` to make this work automagically so weird that it doesn't.

Don't worry about spending any more time on this - it's not really important. But please leave this unresolved and I might try to figure it out some time, either in this PR or a future one.

Reply:
I've decided that we don't to repeat the exact same docs for everything wrapped and unwrapped. Doing things this way round (`capture` and then `__wrapped__`) is fine compared to the opposite direction. The captured version is the more important one to have docstrings, and I've added a note in the API docs to explain what `vizro.figures.library` is.